pkg/util/platformworkloadidentity/platformworkloadidentityrolesbyversion.go

```diff
@@ -86,6 +105,24 @@ func (service *PlatformWorkloadIdentityRolesByVersionService) GetPlatformWorkloa
 	return platformWorkloadIdentityRolesByRoleName
 }
 
+// Check if any required platform identity is missing
+func (service *PlatformWorkloadIdentityRolesByVersionService) MatchesPlatformWorkloadIdentityRoles(oc *api.OpenShiftCluster) bool {
```

Comment:
Since platformIdentity field is a map, there will always be unique key and the combination of length and keys would suffice.
Since it is not clear here, will add test cases and also comments to make it clear.

Reply:
Todo: while printing the error message, the order must be alphabetical to be consistent.

Reply:
Added comments, message sorting is handled in dependent PR: https://github.com/Azure/ARO-RP/pull/3966